crates/cli/src/build.rs

```diff
@@ -332,16 +317,31 @@ pub fn build_file<'a>(
     let link_start = Instant::now();
     let problems = match (linking_strategy, link_type) {
         (LinkingStrategy::Surgical, _) => {
-            roc_linker::link_preprocessed_host(target, &host_input_path, app_o_file, &binary_path);
+            roc_linker::link_preprocessed_host(
+                target,
+                &host_input_path,
+                &roc_app_bytes,
+                &binary_path,
+            );
+
             problems
         }
         (LinkingStrategy::Additive, _) | (LinkingStrategy::Legacy, LinkType::None) => {
             // Just copy the object file to the output folder.
             binary_path.set_extension(app_extension);
-            std::fs::copy(app_o_file, &binary_path).unwrap();
+            std::fs::write(&binary_path, &*roc_app_bytes).unwrap();
             problems
         }
         (LinkingStrategy::Legacy, _) => {
+            let app_o_file = Builder::new()
+                .prefix("roc_app")
+                .suffix(&format!(".{}", app_extension))
+                .tempfile()
+                .map_err(|err| todo!("TODO Gracefully handle tempfile creation error {:?}", err))?;
+            let app_o_file = app_o_file.path();
```

Comment:
now we only actually write the `app.o` to disk when using the surgical linker

Reply:
You mean legacy linker. Awesome. Also, don't we have an option to not link where we should also dump the `.o` to disk?

Reply:
yes, it is covered by this branch
```
        (LinkingStrategy::Additive, _) | (LinkingStrategy::Legacy, LinkType::None) => {
            // Just copy the object file to the output folder.
            binary_path.set_extension(app_extension);
            std::fs::write(&binary_path, &*roc_app_bytes).unwrap();
            problems
        }
```